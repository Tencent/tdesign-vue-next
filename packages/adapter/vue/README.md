# 适配过程记录

## 1. vue
1. instance

```ts
props: innerInstance?.$props
```
## 2. hooks
- useIcon