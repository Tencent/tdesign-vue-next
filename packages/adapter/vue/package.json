{
  "name": "@td/adapter-vue",
  "version": "1.0.0",
  "description": "",
  "author": "",
  "license": "ISC",
  "keywords": [],
  "main": "index.ts",
  "module": "index.ts",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {
    "@td/ts-config": "workspace:^"
  }
}
