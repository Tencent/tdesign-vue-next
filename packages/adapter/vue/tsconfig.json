{
  "extends": "@td/ts-config/vue3.json",
  "compilerOptions": {
    // 应该设置为 true 的，但因 adapter 导出 type，所以设置为 false 了 
    "verbatimModuleSyntax": false,
  }
}
