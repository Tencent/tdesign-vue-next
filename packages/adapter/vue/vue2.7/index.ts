import Vue, { h as H, getCurrentInstance as getCurrentInstanceInner } from 'vue';
import type { Ref, VNode, VueConstructor } from 'vue';
import { clone, lowerFirst, merge, omit } from 'lodash-es';
import Teleport from './components/Teleport';

function getCurrentInstance() {
  const innerInstance = getCurrentInstanceInner()?.proxy;
  const instance = innerInstance
    ? {
        ...innerInstance,
        props: innerInstance.$props,
        vnode: innerInstance.$vnode,
        emit: innerInstance.$emit.bind(innerInstance),
        slots: innerInstance.$scopedSlots,
      }
    : undefined;

  return instance;
}

function getVNode(node: VNode) {
  return {
    ...node,
    props: node.componentOptions?.propsData,
    children: { default: () => node.componentOptions?.children },
  };
}

function cloneVNode(node: VNode, extraProps: Record<string, string | number | Array<string | number>>) {
  return merge(clone(node), extraProps);
}

export interface JSXRenderContext {
  defaultNode?: VNode | string;
  params?: Record<string, any>;
  // 是否不打印 LOG
  silent?: boolean;
}

export type OptionsType = VNode | JSXRenderContext | string;

function isVNode(obj: OptionsType) {
  const vNode = H('span', '');
  const VNode = vNode.constructor;
  return obj instanceof VNode;
}

const Transition = Vue.component('Transition');
const TransitionGroup = Vue.component('TransitionGroup');

function createApp(rootComp: VueConstructor, rootProps: Record<string, any> = {}) {
  const comp = Vue.extend(rootComp);
  const ins = new comp({ propsData: rootProps });
  const mount = (el: string | HTMLElement) => {
    ins.$mount(el);
    return ins;
  };
  return { ...ins, mount };
}

function pluginInstall(app: VueConstructor, plugin: Plugin, ...paths: string[]) {
  let lastObject = app.prototype;
  for (let i = 0; i < paths.length - 1; i++) {
    const path = paths[i];
    if (!path) {
      throw new Error(`Path ${i} is empty`);
    }
    if (!(path in lastObject)) {
      lastObject[path] = {};
    }
    lastObject = lastObject[path];
  }
  const lastPath = paths[paths.length - 1];
  if (!lastPath) {
    throw new Error('Last path is empty');
  }
  lastObject[lastPath] = plugin;
}

// vue23:?? 临时写的方法，待讨论
function createElement(type: string, props: Record<string, any> = {}, children?: any) {
  const innerData = omit(props, ['key', 'ref', 'class', 'style']);
  const event: Record<string, any> = {};
  const data: Record<string, any> = {};

  for (const key in innerData) {
    if (key.startsWith('on')) {
      event[lowerFirst(key.slice(2))] = innerData[key];
    } else {
      data[key] = innerData[key];
    }
  }

  return H(type, {
    key: props.key,
    ref: props.ref,
    class: props.class,
    style: props.style,
    attrs: data,
    props: data,
    on: event,
  }, children);
}

const isVue3 = false;
const isVue2 = true;

export * from 'vue';
export {
  getVNode,
  getCurrentInstance,
  H,
  cloneVNode,
  isVNode,
  Teleport,
  Transition,
  TransitionGroup,
  createApp,
  createElement,
  pluginInstall,
  isVue3,
  isVue2,
};

export default Vue;
