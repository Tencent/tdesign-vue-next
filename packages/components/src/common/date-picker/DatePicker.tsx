import { computed, defineComponent, watch } from '@td/adapter-vue';
import dayjs from 'dayjs';
import { isDate, isFunction } from 'lodash-es';

import type { DateValue, TdDatePickerProps } from '@td/intel/components/date-picker/type';
import { useConfig, useDisabled, usePrefixClass, useTNodeJSX } from '@td/adapter-hooks';
import { formatDate, formatTime, getDefaultFormat, parseToDayjs } from '@td/shared/_common/js/date-picker/format';
import { addMonth, covertToDate, extractTimeObj, subtractMonth } from '@td/shared/_common/js/date-picker/utils';
import props from '@td/intel/components/date-picker/props';
import TSelectInput from '../select-input';
import useSingle from './hooks/useSingle';
import TSinglePanel from './panel/SinglePanel';

export default defineComponent({
  name: 'TDatePicker',

  props: {
    ...props,
  },

  setup(props) {
    const COMPONENT_NAME = usePrefixClass('date-picker');

    const {
      inputValue,
      popupVisible,
      inputProps,
      popupProps,
      isHoverCell,
      cacheValue,
      value,
      year,
      month,
      time,
      inputRef,
      onChange,
    } = useSingle(props);

    // vue23:!
    const disabled = useDisabled();
    const renderTNodeJSX = useTNodeJSX();
    const { globalConfig } = useConfig('datePicker');

    const formatRef = computed(() =>
      getDefaultFormat({
        mode: props.mode,
        format: props.format,
        valueType: props.valueType,
        enableTimePicker: props.enableTimePicker,
      }),
    );
    const valueDisplayParams = computed(() => {
      return {
        value: value.value,
        displayValue: inputValue.value,
      };
    });

    watch(popupVisible, (visible) => {
      const dateValue
        // Date 属性不再 parse，避免 dayjs 处理成 Invalid
        = value.value && !isDate(value.value)
          ? covertToDate(value.value as string, formatRef.value?.valueType)
          : value.value;

      cacheValue.value = formatDate(dateValue, {
        format: formatRef.value.format,
      });
      inputValue.value = formatDate(dateValue, {
        format: formatRef.value.format,
      });

      // 面板展开重置数据
      if (visible) {
        year.value = parseToDayjs(value.value, formatRef.value.format).year();
        month.value = parseToDayjs(value.value, formatRef.value.format).month();
        time.value = formatTime(value.value, formatRef.value.format, formatRef.value.timeFormat, props.defaultTime);
      } else {
        isHoverCell.value = false;
      }
    });

    // 日期 hover
    function onCellMouseEnter(date: Date) {
      isHoverCell.value = true;
      inputValue.value = formatDate(date, {
        format: formatRef.value.format,
      });
    }

    // 日期 leave
    function onCellMouseLeave() {
      isHoverCell.value = false;
      inputValue.value = formatDate(cacheValue.value, {
        format: formatRef.value.format,
      });
    }

    // 日期点击
    function onCellClick(date: Date) {
      isHoverCell.value = false;
      // date 模式自动切换年月
      if (props.mode === 'date') {
        year.value = date.getFullYear();
        month.value = date.getMonth();
      }
      if (props.enableTimePicker) {
        cacheValue.value = formatDate(date, {
          format: formatRef.value.format,
        });
      } else {
        onChange?.(
          formatDate(date, {
            format: formatRef.value.format,
            targetFormat: formatRef.value.valueType,
          }) as DateValue,
          {
            dayjsValue: parseToDayjs(date, formatRef.value.format),
            trigger: 'pick',
          },
        );
        popupVisible.value = false;
      }

      props.onPick?.(date);
    }

    // 头部快速切换
    function onJumperClick({ trigger }: { trigger: string }) {
      const monthCountMap = { date: 1, week: 1, month: 12, quarter: 12, year: 120 };
      const monthCount = monthCountMap[props.mode] || 0;

      const current = new Date(year.value, month.value);

      let next = null;
      if (trigger === 'prev') {
        next = subtractMonth(current, monthCount);
      } else if (trigger === 'current') {
        next = new Date();
      } else if (trigger === 'next') {
        next = addMonth(current, monthCount);
      }

      const nextYear = next.getFullYear();
      const nextMonth = next.getMonth();

      year.value = nextYear;
      month.value = nextMonth;
    }

    // timePicker 点击
    function onTimePickerChange(val: string) {
      time.value = val;

      const { hours, minutes, seconds, milliseconds, meridiem } = extractTimeObj(val);

      // am pm 12小时制转化 24小时制
      let nextHours = hours;
      if (/am/i.test(meridiem) && nextHours === 12) {
        nextHours -= 12;
      }
      if (/pm/i.test(meridiem) && nextHours < 12) {
        nextHours += 12;
      }
      const currentDate = !dayjs(inputValue.value as string, formatRef.value.format).isValid()
        ? dayjs()
        : dayjs(inputValue.value as string, formatRef.value.format);
      const nextDate = currentDate.hour(nextHours).minute(minutes).second(seconds).millisecond(milliseconds).toDate();
      inputValue.value = formatDate(nextDate, {
        format: formatRef.value.format,
      });
      cacheValue.value = formatDate(nextDate, {
        format: formatRef.value.format,
      });

      props.onPick?.(nextDate);
    }

    // 确定
    function onConfirmClick({ e }: { e: MouseEvent }) {
      const nextValue = formatDate(inputValue.value, {
        format: formatRef.value.format,
      });
      if (nextValue) {
        props?.onConfirm?.({ date: dayjs(nextValue as string).toDate(), e });
        onChange?.(
          formatDate(inputValue.value, {
            format: formatRef.value.format,
            targetFormat: formatRef.value.valueType,
          }) as DateValue,
          {
            dayjsValue: parseToDayjs(inputValue.value as string, formatRef.value.format),
            trigger: 'confirm',
          },
        );
      } else {
        inputValue.value = formatDate(value.value, {
          format: formatRef.value.format,
        });
      }
      popupVisible.value = false;
    }

    // 预设
    function onPresetClick(presetValue: DateValue | (() => DateValue)) {
      const presetVal = isFunction(presetValue) ? presetValue() : presetValue;
      onChange?.(
        formatDate(presetVal, {
          format: formatRef.value.format,
          targetFormat: formatRef.value.valueType,
        }) as DateValue,
        {
          dayjsValue: parseToDayjs(presetVal, formatRef.value.format),
          trigger: 'preset',
        },
      );

      popupVisible.value = false;
    }

    function onYearChange(nextYear: number) {
      year.value = nextYear;
    }

    function onMonthChange(nextMonth: number) {
      month.value = nextMonth;
    }

    const panelProps = computed(() => ({
      value: cacheValue.value as string,
      year: year.value,
      month: month.value,
      format: formatRef.value.format,
      mode: props.mode,
      presets: props.presets,
      time: time.value as string,
      disableDate: props.disableDate,
      firstDayOfWeek: props.firstDayOfWeek,
      timePickerProps: props.timePickerProps,
      enableTimePicker: props.enableTimePicker,
      presetsPlacement: props.presetsPlacement,
      popupVisible: popupVisible.value,
      onCellClick,
      onCellMouseEnter,
      onCellMouseLeave,
      onJumperClick,
      onConfirmClick,
      onPresetClick,
      onYearChange,
      onMonthChange,
      onTimePickerChange,
      onPanelClick: () => inputRef.value?.focus?.(),
    }));

    return () => (
      <div class={COMPONENT_NAME.value}>
        <TSelectInput
          disabled={disabled.value}
          value={inputValue.value}
          status={props.status}
          tips={props.tips}
          clearable={props.clearable}
          popupProps={popupProps.value}
          inputProps={inputProps.value}
          placeholder={props.placeholder || globalConfig.value.placeholder[props.mode]}
          popupVisible={popupVisible.value}
          valueDisplay={() => renderTNodeJSX('valueDisplay', { params: valueDisplayParams.value })}
          {...(props.selectInputProps as TdDatePickerProps['selectInputProps'])}
          panel={() => <TSinglePanel {...panelProps.value} />}
        />
      </div>
    );
  },
});
