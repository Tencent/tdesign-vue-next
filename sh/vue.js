const fs = require('fs');
const path = require('path');

// 定义源文本的正则表达式
const src_regex = /from 'vue';/g;

// 定义要搜索的目录
const dirs = ['packages/components/common/src', 'packages/components/vue3/src'];

// 定义处理文件的函数
function processFiles(directory) {
    fs.readdirSync(directory, { withFileTypes: true }).forEach(file => {
        const filePath = path.join(directory, file.name);
        if (file.isDirectory()) {
            processFiles(filePath);
        } else if (file.isFile() && (filePath.endsWith('.js') || filePath.endsWith('.ts') || filePath.endsWith('.tsx'))) {
            let content = fs.readFileSync(filePath, 'utf8');
            content = content.replace(src_regex, "from '@td/adapter-vue';");
            fs.writeFileSync(filePath, content, 'utf8');
        }
    });
}

// 遍历目录并处理文件
dirs.forEach(dir => processFiles(dir));