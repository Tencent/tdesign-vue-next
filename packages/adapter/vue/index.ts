import * as Vue from '@adapter/vue';

export * from '@adapter/vue';

export default Vue;
