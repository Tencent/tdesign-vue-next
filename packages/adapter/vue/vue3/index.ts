import * as Vue from 'vue';
import { h as H, getCurrentInstance } from 'vue';
import type { App, Plugin, VNode } from 'vue';

function getVNode(node: VNode) {
  return node;
}

function pluginInstall(app: App, plugin: Plugin, ...paths: string[]) {
  let lastObject = app.config.globalProperties;
  for (let i = 0; i < paths.length - 1; i++) {
    const path = paths[i];
    if (!path) {
      throw new Error(`Path ${i} is empty`);
    }
    if (!(path in lastObject)) {
      lastObject[path] = {};
    }
    lastObject = lastObject[path];
  }
  const lastPath = paths[paths.length - 1];
  if (!lastPath) {
    throw new Error('Last path is empty');
  }
  lastObject[lastPath] = plugin;
}

const createElement = H;

const isVue3 = true;
const isVue2 = false;

export * from 'vue';
export {
  getCurrentInstance,
  getVNode,
  createElement,
  H,
  pluginInstall,
  isVue3,
  isVue2,
};

export default Vue;
