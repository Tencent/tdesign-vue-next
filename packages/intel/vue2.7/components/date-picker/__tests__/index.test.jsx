import { mount } from '@vue/test-utils';
import MockDate from 'mockdate';
import { nextTick } from 'vue';
import { BrowseIcon, LockOnIcon } from 'tdesign-icons-vue';
import dayjs from 'dayjs';
import DatePicker from '@/src/date-picker/index.ts';

// 固定时间，当使用 new Date() 时，返回固定时间，防止“当前时间”的副作用影响，导致 snapshot 变更，mockdate 插件见 https://github.com/boblauer/MockDate
MockDate.set('2020-12-28');

// every component needs four parts: props/events/slots/functions.
describe('DatePicker', () => {
  // test props api
  describe(':props', () => {
    it('', () => {
      const wrapper = mount({
        render() {
          return <DatePicker />;
        },
      });
      expect(wrapper.exists()).toBe(true);
    });
  });

  it(':value', () => {
    const wrapper = mount({
      render() {
        return <DatePicker value={'1998-11-11'} />;
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it(':mode', () => {
    const wrapper = mount({
      render() {
        return <DatePicker mode={'year'} />;
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it(':range', () => {
    const wrapper = mount({
      render() {
        const testRange = [
          '2018-08', // new Date(2017, 7)
          '2028-04', // new Date(2027, 3)
        ];
        return <DatePicker value={testRange} range />;
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('clearable', async () => {
    const wrapper = mount({
      render() {
        return <DatePicker defaultValue={'2022-09-14'} clearable></DatePicker>;
      },
    });
    const input = wrapper.find('.t-input');
    await input.trigger('mouseenter');
    await nextTick();
    expect(wrapper.find('.t-input__suffix-clear').exists()).toBeTruthy();
  });

  it('disabled', () => {
    const wrapper = mount({
      render() {
        return <DatePicker disabled />;
      },
    });
    const inputInner = wrapper.find('.t-input__inner');
    expect(inputInner.element.disabled).toBe(true);
  });

  it('enableTimePicker', async () => {
    const wrapper = mount({
      render() {
        return <DatePicker enableTimePicker />;
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    expect(document.querySelector('.t-date-picker__panel-time')).not.toBe(null);
  });

  it('firstDayOfWeek', async () => {
    const wrapper = mount(DatePicker, {
      propsData: {
        firstDayOfWeek: 3,
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    const weekElement = document.querySelector('.t-date-picker__table table thead tr th');
    expect(weekElement.innerHTML).toEqual('一');
  });

  it('format', async () => {
    // defaultValue
    const wrapper = mount({
      render() {
        return <DatePicker defaultValue={'2022-09-14 16:30:20'} format={'YYYY-MM-DD'} />;
      },
    });
    const inputElement1 = wrapper.find('.t-input__inner');
    expect(inputElement1.element.value).toEqual('2022-09-14');

    // value
    const wrapper2 = mount({
      render() {
        return <DatePicker value={'2022-09-14 16:30:20'} format={'YYYY-MM-DD'} />;
      },
    });
    const inputElement2 = wrapper2.find('.t-input__inner');
    expect(inputElement2.element.value).toEqual('2022-09-14');
  });

  it('inputProps', () => {
    const wrapper = mount(DatePicker, {
      propsData: {
        inputProps: { inputClass: 'test-inputClass' },
      },
    });
    expect(wrapper.find('.t-input').classes()).toContain('test-inputClass');
  });

  it('mode week', async () => {
    const wrapper = mount({
      render() {
        return <DatePicker mode={'week'} value={'2022-37th'} />;
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    expect(document.querySelector('.t-date-picker__panel-week')).not.toBe(null);
  });

  it('mode month', async () => {
    const wrapper = mount({
      render() {
        return <DatePicker mode={'month'} />;
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    expect(document.querySelector('.t-date-picker__panel-month')).not.toBe(null);
  });

  it('mode quarter', async () => {
    const wrapper = mount({
      render() {
        return <DatePicker mode={'quarter'} />;
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    expect(document.querySelector('.t-date-picker__panel-quarter')).not.toBe(null);
  });

  it('mode year', async () => {
    const wrapper = mount({
      render() {
        return <DatePicker mode={'year'} />;
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    expect(document.querySelector('.t-date-picker__panel-year')).not.toBe(null);
  });

  it('placeholder', () => {
    const wrapper = mount({
      render() {
        return <DatePicker />;
      },
    });
    expect(wrapper.find('input').element.placeholder).toEqual('请选择日期');
  });

  it('popupProps', async () => {
    const wrapper = mount({
      render() {
        return <DatePicker popupProps={{ showArrow: true }} />;
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    expect(document.querySelector('.t-popup__arrow')).not.toBe(null);
  });

  it('prefixIcon suffixIcon', () => {
    const wrapper = mount({
      render() {
        return <DatePicker prefixIcon={() => <BrowseIcon />} suffixIcon={() => <LockOnIcon />} />;
      },
    });
    expect(wrapper.find('.t-icon-browse')).not.toBe(null);
    expect(wrapper.find('.t-icon-lock-on')).not.toBe(null);
  });

  it('presets bottom top', async () => {
    const presets = {
      昨天: dayjs().subtract(1, 'day').format('YYYY-MM-DD'),
    };
    const wrapper = mount({
      render() {
        return <DatePicker presets={presets} />;
      },
    });
    wrapper.find('.t-input').trigger('click');
    await nextTick();
    expect(document.querySelector('.t-date-picker__presets .t-button__text').innerHTML).toContain('昨天');
  });

  test('status', () => {
    const wrapper = mount({
      render() {
        return <DatePicker status={'warning'} />;
      },
    });
    expect(wrapper.find('.t-input').classes()).toContain('t-is-warning');
  });

  // test exposure function
  describe('function', () => {
    it('onBlur', async () => {
      const blurFn = vi.fn();
      const wrapper = mount({
        render() {
          return <DatePicker allowInput onBlur={blurFn} />;
        },
      });
      const InputDom = wrapper.find('input');
      InputDom.trigger('blur');
      expect(blurFn).toBeCalledTimes(1);
    });
  });
});
