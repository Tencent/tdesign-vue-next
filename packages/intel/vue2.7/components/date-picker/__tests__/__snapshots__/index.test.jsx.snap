// Vitest Snapshot v1

exports[`DatePicker > :mode 1`] = `
<div
  class="t-date-picker"
>
  <div
    class="t-select-input t-select-input--empty"
  >
    <div
      class="t-input__wrap"
    >
      <div
        class="t-input t-is-readonly t-input--prefix t-input--suffix"
      >
        <div
          class="t-input__prefix"
        />
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请选择年份"
          readonly="readonly"
          type="text"
          unselectable="on"
        />
        <span
          class="t-input__suffix t-input__suffix-icon"
        >
          <svg
            class="t-icon t-icon-calendar"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M6 4V1.5h2V4h8V1.5h2V4h4v18H2V4h4zM4 6v3h16V6H4zm16 5H4v9h16v-9z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker > :range 1`] = `
<div
  class="t-date-picker"
  range="true"
>
  <div
    class="t-select-input"
  >
    <div
      class="t-input__wrap"
    >
      <div
        class="t-input t-is-readonly t-input--prefix t-input--suffix"
      >
        <div
          class="t-input__prefix"
        />
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请选择日期"
          readonly="readonly"
          type="text"
          unselectable="on"
        />
        <span
          class="t-input__suffix t-input__suffix-icon"
        >
          <svg
            class="t-icon t-icon-calendar"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M6 4V1.5h2V4h8V1.5h2V4h4v18H2V4h4zM4 6v3h16V6H4zm16 5H4v9h16v-9z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker > :value 1`] = `
<div
  class="t-date-picker"
>
  <div
    class="t-select-input"
  >
    <div
      class="t-input__wrap"
    >
      <div
        class="t-input t-is-readonly t-input--prefix t-input--suffix"
      >
        <div
          class="t-input__prefix"
        />
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请选择日期"
          readonly="readonly"
          type="text"
          unselectable="on"
        />
        <span
          class="t-input__suffix t-input__suffix-icon"
        >
          <svg
            class="t-icon t-icon-calendar"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M6 4V1.5h2V4h8V1.5h2V4h4v18H2V4h4zM4 6v3h16V6H4zm16 5H4v9h16v-9z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
